contracts/contracts/MessageProcessor.sol

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import {AccQueue} from "./trees/AccQueue.sol";
+import {IMACI} from "./IMACI.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {Poll} from "./Poll.sol";
+import {SnarkCommon} from "./crypto/SnarkCommon.sol";
+import {CommonUtilities} from "./Utility.sol";
+import {Verifier} from "./crypto/Verifier.sol";
+import {VkRegistry} from "./VkRegistry.sol";
+
+contract MessageProcessor is Ownable, SnarkCommon, CommonUtilities {
+
+    string constant ERROR_VOTING_PERIOD_PASSED = "ProcessE01";
```

Comment:
As detailed in the other comments, I think it'd be great to move to custom errors and replace require statements with if checks

Reply:
done